### PR TITLE
🧪 **Test-Driven Development: Implementation of Solver Class**

### PR DESCRIPTION
**Pull request by egichuhis**
This pull request introduces the `Solver` class with three methods: `factorial`, `reverse`, and `fizzbuzz`. The implementation follows the Test-Driven Development (TDD) approach, ensuring that unit tests are created first and the code is subsequently implemented to pass those tests.

### Implemented Features:

1. **Method: `factorial`**
   - Created a method that calculates the factorial of a given integer `N`.
   - Handles the special case where the factorial of 0 is 1.
   - Raises an exception for negative integers.

2. **Method: `reverse`**
   - Implemented a method that reverses a given string `word`.

3. **Method: `fizzbuzz`**
   - Created a method that applies the FizzBuzz logic.
   - Returns "fizz" when `N` is divisible by 3.
   - Returns "buzz" when `N` is divisible by 5.
   - Returns "fizzbuzz" when `N` is divisible by both 3 and 5.
   - Returns `N` as a string for any other case.

### Test Suite:

- Comprehensive test suite for each method to ensure correctness.
- Utilized various test scenarios, including edge cases and typical cases.

### Commit History:

- Commits demonstrate the incremental development process following TDD.
- Each commit is associated with a specific test case and its corresponding implementation.

### Usage Instructions:

1. Ensure you have a Ruby environment set up.
2. Run the test suite to verify the correctness of the `Solver` class. Using `bundle exec rspec` command

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,112 @@
...
+
+[![Star](https://img.shields.io/github/stars/agneta1991/tdd?style=social)](https://github.com/Ochiengsteven/TDD)
+
+<!-- ACKNOWLEDGEMENTS -->
+## 🙏 Acknowledgments <a name="acknowledgements"></a>
+We would like to express our gratitude to the contributors and anyone who uses or supports this project.
+
+<!-- LICENSE -->
+## 📝 License <a name="license"></a>
+This project is [MIT](./LICENSE) licensed.
```
Comment (mikethreels) [inline README.md]: - [ ] the link to the MIT license is not working as expected, please make sure it is working as shown in the readme template provided by Microverse.

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Team, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Good job writing the tests

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by mikethreels — APPROVED** [review]
Hi Team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- good job implementing the requested changes.
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
